### PR TITLE
OBSDOCS-1231/OBSDOCS-1058: Update docs about instrumentation for multi-container pods

### PR DESCRIPTION
**Pull request by max-cx**
:warning: DO NOT MERGE! `do-not-merge/hold`

:warning: This PR is tied to a product release date, which is currently set to **December 10**.

- If the merge review gets delayed, you're welcome to ping `mleonov` on Slack.
- If you have any questions, ping `mleonov` on Slack.

Signed-off-by: Israel Blancas <iblancasa@gmail.com>, PR based on https://github.com/openshift/openshift-docs/pull/77259/

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16, 4.17, 4.18

Issue:

https://issues.redhat.com/browse/OBSDOCS-1231
https://issues.redhat.com/browse/OBSDOCS-1058
https://issues.redhat.com/browse/TRACING-4213
https://issues.redhat.com/browse/TRACING-4270

Link to docs preview:

https://78020--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-instrumentation.html#otel-multi-container-pods-multi-instrumentations_otel-configuration-of-instrumentation

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/otel-config-instrumentation.adoc [hunk 1]
```diff
@@ -344,3 +344,55 @@ instrumentation.opentelemetry.io/container-names: "<container_1>,<container_2>"
...
+[source,yaml]
+----
+instrumentation.opentelemetry.io/python-container-names: "<container_1>,<container_2>"
+----
+
+.Multi instrumentation for DotNet applications
+[source,yaml]
+----
+instrumentation.opentelemetry.io/go-container-names: "<container_1>"
+----
```
Comment (max-cx) [inline modules/otel-config-instrumentation.adoc:375]: Should this be one line for DotNet and one line for Go?

File: modules/otel-config-instrumentation.adoc [hunk 2]
```diff
@@ -344,3 +344,46 @@ instrumentation.opentelemetry.io/container-names: "<container_1>,<container_2>"
...
+[id="otel-multi-container-pods-multi-instrumentations_{context}"]
+=== Multi-container pods with multiple instrumentations
+
+Injecting instrumentation for an application language to one or more containers in a multi-container pod requires the following annotation:
+
+[source,yaml]
+----
+instrumentation.opentelemetry.io/<application_language>-container-names: "<container_1>,<container_2>" # <1>
+----
+<1> You can only inject instrumentation for one language per container. For the list of supported `<application_language>` values, see the following table.
```
Comment (mburke5678) [inline modules/otel-config-instrumentation.adoc:357]: "only" should be as close as possible to the term it modifies.

```suggestion
<1> You can inject instrumentation for only one language per container. For the list of supported `<application_language>` values, see the following table.
```

**Comment by mburke5678** [comment]
@max-cx One tiny nit. Otherwise LGTM

File: modules/otel-config-instrumentation.adoc (hunk 2 above)
Comment (max-cx) [reply modules/otel-config-instrumentation.adoc:357]: Thank you. Fixed.

**Review by IshwarKanse — APPROVED** [review]

**Review by ShaunaDiaz — COMMENTED** [review]
/lgtm

**Pull request merged**